docs-next/pages/apis/hooks.mdx

```diff
@@ -2,6 +2,455 @@ import { Markdown } from '../../components/Page';
+          hooks: {
+            beforeDelete: async ({ listKey, fieldPath, operation, existingItem, context }) => {
+              /* ... */
+            },
+          },
+        }),
+      },
+    }),
+  }),
+});
+```
+
+### afterDelete
+
+The `afterDelete` function is used to perform side effects after the data for a `delete` operation has been removed from the database.
+
+| Argument       | Description                                                                                                                                                  |
+| :------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `listKey`      | The key of the list being operated on.                                                                                                                       |
+| `fieldPath`    | The path of the field being operated on (field hooks only).                                                                                                  |
+| `operation`    | The operation being performed (`'delete'`).                                                                                                                  |
+| `existingItem` | The value of the item, now deleted. This object is an unresolved list item. See the [list item API](./list-items) for more details on unresolved list items. |
+| `context`      | The [`KeystoneContext`](./context) object of the originating GraphQL operation.                                                                              |
+
+```typescript
+import { config, createSchema, list } from '@keystone-next/keystone/schema';
+import { text } from '@keystone-next/fields';
+
+export default config({
+  lists: createSchema({
+    ListName: list({
+      hooks: {
+        afterDelete: async ({ listKey, operation, existingItem, context }) => {
+          /* ... */
+        },
+      },
+      fields: {
+        fieldName: text({
+          hooks: {
+            afterDelete: async ({ listKey, fieldPath, operation, existingItem, context }) => {
+              /* ... */
+            },
+          },
+        }),
+      },
+    }),
+  }),
+});
+```
+
+## Resolved data stages
+
+Create and update operations take a `data` value for a single item from the GraphQL input and then perform a number of _data resolving_ steps before writing the final value to the database.
+
+At each stage of the data resolving process, the value of `resolvedData` can be modifed or augmented.
+The final value of `resolvedData` is the value that will be validated and saved to the database.
+
+The data resolving steps are applied in the following order:
+
+1. Initialisation: Set the value of `resolvedData` to the `data` input value from the GraphQL mutation.
+2. Defaults (built in): Any fields which have a default value and are `undefined` in `resolvedData` will be set to their default value.
```

Comment:
I believe this is only relevant for `create` operations? above, this section is scoped to "Create and update operations" so lets call this exception out if I'm right